ide/git/test/unit/src/org/netbeans/modules/git/AbstractGitTestCase.java

```diff
@@ -233,41 +265,36 @@ protected final void runExternally (File workdir, List<String> command) throws E
         ProcessBuilder pb = new ProcessBuilder(command);
         pb.directory(workdir);
         Process p = pb.start();
-        final BufferedReader outReader = new BufferedReader(Channels.newReader(Channels.newChannel(p.getInputStream()), "UTF-8"));
-        final BufferedReader errReader = new BufferedReader(Channels.newReader(Channels.newChannel(p.getErrorStream()), "UTF-8"));
-        final List<String> output = new LinkedList<String>();
-        final List<String> err = new LinkedList<String>();
-        Thread t1 = new Thread(new Runnable() {
-            @Override
-            public void run () {
+        final List<String> err;
+        try (BufferedReader outReader = new BufferedReader(Channels.newReader(Channels.newChannel(p.getInputStream()), "UTF-8"))) {
+            BufferedReader errReader = new BufferedReader(Channels.newReader(Channels.newChannel(p.getErrorStream()), "UTF-8"));
+            @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+            final List<String> output = new LinkedList<>();
+            err = new LinkedList<>();
+            Thread t1 = new Thread(() -> {
                 try {
                     for (String line = outReader.readLine(); line != null; line = outReader.readLine()) {
                         output.add(line);
                     }
                 } catch (IOException ex) {
-                    
+                    // Ignore
```

Comment:
wondering if it would be better to fail the test on IOE?
```
    output.add(ex.getMessage());
```
adding the msg to the list would fail the test a few lines later